pkg/storage/access_token_file_test.go

```diff
@@ -0,0 +1,149 @@
+package storage
+
+import (
+	storageConfig "github.com/cyberark/conjur-authn-k8s-client/pkg/storage/config"
+	. "github.com/smartystreets/goconvey/convey"
+	"os"
+	"reflect"
+	"testing"
+)
+
+type ProxyHandlerTokenFile struct {
+	AccessToken AccessTokenHandler
+}
+
+func TestAccessTokenFile(t *testing.T) {
+	var config storageConfig.Config
+	config.StoreType = storageConfig.K8S
+	config.TokenFilePath = "/tmp/accessTokenFile1"
+	var tokenInFile, _ = NewAccessTokenFile(config)
+
+	Convey("Read", t, func() {
+		Convey("Returns true if Data output as expected", func() {
+			dataActual := []byte{'t', 'e', 's', 't'}
+			err := tokenInFile.Write(dataActual)
+			So(err, ShouldEqual, nil)
+
+			dataExpected, err := tokenInFile.Read()
+			eq := reflect.DeepEqual(dataActual, dataExpected)
+
+			So(err, ShouldEqual, nil)
+			So(eq, ShouldEqual, true)
+		})
+
+		Convey("Returns error if Data is nil", func() {
+			tokenInFile.Data = nil
+			_, err := tokenInFile.Read()
+
+			So(err.Error(), ShouldEqual, "error reading access token, reason: data is empty")
+		})
+	})
+
+	Convey("Write", t, func() {
+		Convey("Returns no error if Data input is not nil", func() {
+			dataActual := []byte{'t', 'e', 's', 't'}
+			tokenInFile.TokenFilePath = "/tmp/accessTokenFileWrite1"
+			err := tokenInFile.Write(dataActual)
+
+			So(err, ShouldEqual, nil)
+			// Check if file exits
+			_, err = os.Stat("/tmp/accessTokenFileWrite1")
+			So(err, ShouldEqual, nil)
+		})
+
+		Convey("Returns error if Data input is nil", func() {
+			err := tokenInFile.Write(nil)
+
+			So(err.Error(), ShouldEqual, "error writing access token, reason: data is empty")
+		})
+
+		Convey("Returns no error if file already exists", func() {
+			dataActual := []byte{'t', 'e', 's', 't'}
+			tokenInFile.TokenFilePath = "/tmp/accessTokenFileWrite2"
+			err := tokenInFile.Write(dataActual)
+			So(err, ShouldEqual, nil)
+
+			// Check if file exits
+			_, err = os.Stat("/tmp/accessTokenFileWrite1")
+			So(err, ShouldEqual, nil)
+
+			// Second time
+			dataActual = []byte{'t', 'e', 's', 't', '2'}
+			err = tokenInFile.Write(dataActual)
+			So(err, ShouldEqual, nil)
+
+			// Check if file exits
+			_, err = os.Stat("/tmp/accessTokenFileWrite1")
+			So(err, ShouldEqual, nil)
+
+			// Test we are reading the new Data
+			dataExpected, err := tokenInFile.Read()
+			eq := reflect.DeepEqual(dataActual, dataExpected)
+
+			So(err, ShouldEqual, nil)
+			So(eq, ShouldEqual, true)
+		})
+	})
+
+	Convey("Delete", t, func() {
+		Convey("Returns no error after read write and delete", func() {
+			dataActual := []byte{'t', 'e', 's', 't'}
+			tokenInFile.TokenFilePath = "/tmp/accessTokenFileDel1"
+			err := tokenInFile.Write(dataActual)
+			So(err, ShouldEqual, nil)
+
+			_, err = tokenInFile.Read()
+			So(err, ShouldEqual, nil)
+
+			err = tokenInFile.Delete()
+			So(err, ShouldEqual, nil)
+
+			// Check that file is not exits
```

Comment:
"check that file does not exist"

Reply:
why wouldn't the file exist after delete func is run? Isn't delete just for deleting _content_ from access token?